src/common/libsubprocess/test/subprocess.c

```diff
@@ -1459,6 +1459,49 @@ void test_state_change (flux_reactor_t *r)
     flux_cmd_destroy (cmd);
 }
 
+void state_change_stopped_cb (flux_subprocess_t *p,
+                              flux_subprocess_state_t state)
+{
+    diag ("state_change_stopped: state = %s",
+          flux_subprocess_state_string (state));
+    if (state == FLUX_SUBPROCESS_STOPPED) {
+        flux_future_t *f = flux_subprocess_kill (p, SIGKILL);
+        ok (true, "subprocess state == STOPPED in state change handler");
+        flux_future_destroy (f);
+    }
+    state_change_cb_count++;
+}
```

Comment:
the way this test is written I don't think its guaranteed that FLUX_SUBPROCESS_STOPPED was seen.  Perhaps instead set a global counter `stopped_count` and update that counter here.  Then in `test_state_change_stopped()` check that `stopped_count == 1`.